src/openMVG/sfm/pipelines/localization/SfM_Localizer.cpp

```diff
@@ -161,7 +155,8 @@ bool SfM_Localizer::RefinePose
   // pose
   sfm_data.poses[0] = pose;
   // intrinsic (the shared_ptr does not take the ownership, will not release the input pointer)
-  sfm_data.intrinsics[0] = std::shared_ptr<cameras::IntrinsicBase>(intrinsics, [](cameras::IntrinsicBase*){});
+  std::shared_ptr<cameras::IntrinsicBase> localIntrinsics(intrinsics->clone());
```

Comment:
But the older version it was just making it shared without copying, isn't it?
As for the shared intrinsics, I agree, but this function just does the refinement of the pose + intrinsics, it should be user (the callee) responsibility to take care of (possible) shared intrinsics with other cameras.